lib/temporal/worker.rb

```diff
@@ -0,0 +1,71 @@
+require 'temporal/sdk/core/core_interface_pb'
+require 'temporal/bridge'
+require 'temporal/runtime'
+require 'temporal/worker/reactor'
+
+module Temporal
+  class Worker
+    class AlreadyStarted < StandardError; end
+
+    def initialize(connection, namespace, task_queue)
+      runtime = Temporal::Runtime.instance
+      runtime.ensure_callback_loop
+
+      @running = false
+      @core_worker = Temporal::Bridge::Worker.create(
+        runtime.core_runtime,
+        connection.core_connection,
+        namespace,
+        task_queue
+      )
+    end
+
+    def run
+      start
+      Reactor.attach
+    end
+
+    def start
+      raise AlreadyStarted if running?
+
+      @running = true
+
+      Reactor.execute do |reactor|
```

Comment:
https://github.com/temporalio/sdk-ruby/pull/38

Reply:
Thanks for looking into this. I've responded on your PR why this might not work as you expect. The main issue is that `core_worker.poll_activity_task` call is non-blocking and this example will just keep spawning polls endlessly

Reply:
Provided the answer here — https://github.com/temporalio/sdk-ruby/pull/38#issuecomment-1219633457